### DISCUSSION TITLE
Recomended Python version?

### DISCUSSION THREAD

**Discussion by skumpi** [squidfunk/mkdocs-material#7319] [Q&A] [answered]
I have a issues and can't run event the minimum startup project. 

 Is there a maximum python verision?

**Comment by MGaetan89** [comment]
According to https://pypi.org/project/mkdocs-material/, the minimal Python version is 3.8

**Reply by skumpi** [reply]
Ok, understand and i'm running 3.12 so that should be fine?

**Reply by alexvoss** [reply]
Running Python 3.12.4 (on MacOS) with no problem. Can you be a bit more specific as to what the issue is?

**Reply by skumpi** [reply]
I have an older mkdocs project i wanted to run on new laptop. 

After first running into issues i reinstalled python and installed py -m pip install mkdocs-material as the only plugin. After that i tried serving the minimal configuration in a new directory. 

It works when i remove the theme configuration, but not with material activated.

<img width="880" alt="image" src="https://github.com/squidfunk/mkdocs-material/assets/1180203/d54b3b9c-6561-42a6-acb5-b2b88dde54aa">


Error:  
  File "C:\Users\JohanEriksson\AppData\Local\Programs\Python\Python312\Lib\site-packages\jinja2\environment.py", line 972, in _load_template
    template = self.loader.load(self, name, self.make_globals(globals))        
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^        
  File "C:\Users\JohanEriksson\AppData\Local\Programs\Python\Python312\Lib\site-packages\jinja2\loaders.py", line 126, in load
    source, filename, uptodate = self.get_source(environment, name)
                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\JohanEriksson\AppData\Local\Programs\Python\Python312\Lib\site-packages\jinja2\loaders.py", line 207, in get_source
    raise TemplateNotFound(template)
jinja2.exceptions.TemplateNotFound: partials/language.html

Added debug zip:
[9.5.27-minimal-start.zip](https://github.com/user-attachments/files/16053570/9.5.27-minimal-start.zip)

**Reply by kamilkrzyskow** [reply]
Windows, currently running 3.12.4 and MkDocs 1.6.0 without issues, also worked fine with 3.11 and 3.10 in the past with MkDocs 1.4+ etc. Share more details about your setup. You can try running `python --version` or `python3 --version`, then do `python -m pip freeze` to see installed packages and then `python -m mkdocs` to run the module.
`mkdocs` and `pip` should be registered as commands separately, but maybe they don't work so run them via `python -m ...`

**Reply by kamilkrzyskow** [reply]
Maybe broken install? Try doing `pip install mkdocs-material --force-reinstall` or the `python -m pip ...` approach.

**Reply by skumpi** [reply]
Thank's tried that several times both before and after python re-install no change. have tried python 3.12.4 and 3.12.1

**Reply by squidfunk** [reply, marked as answer]
Make sure you've installed MkDocs and Material for MkDocs in the same location. It's recommended to use a virtual environment.

**Reply by alexvoss** [reply]
So, just happened to look at the `platform.json` in your reproduction and it occurs to me that your PATH contains a directory `C:\\Users\\JohanEriksson\\AppData\\Roaming\\Python\\Python312\\site-packages` before `C:\\Users\\JohanEriksson\\AppData\\Local\\Programs\\Python\\Python312\\Lib\\site-packages`. Is it conceivable that Jinja2 looks in the first `site-packages` folder it finds and that the partial is not there because it is installed in the 2nd, where Material lives, the one under `AppData`?

**Reply by alexvoss** [reply]
Whether or not what I just noticed has anything to do with it, I think that what @squidfunk suggests is the sane thing to do in any case.

**Reply by kamilkrzyskow** [reply]
Yes, likely 2 separate installations of material in 2 different sources, and 2 different versions, a virtual environment should help.
I also recommend not using Python installed from the App Store, just install it manually via the installer from python.org
Or use MSYS2 with an isolated environment and special packages.

Here is your example with an additional hook, which adds more debug data about the visible templates in Jinja + visible files in the used material installation.

You can run the command with `2>&1` to pipe the logs into a file:
```powershell
python -m mkdocs build > log.txt 2>&1
```

[skumpi-minimal-test.zip](https://github.com/user-attachments/files/16053966/skumpi-minimal-test.zip)
EDIT: I sent broken code by mistake, fixed the link above

**Reply by skumpi** [reply]
Thanks alot for you help, got it working now. Got  it working by using virtual environment like surgested by @squidfunk. 

But since i can't leave well enough alone i also now got it working withouth it by manually unistalling python and empty both the dirs mentioned by @alexvoss and remoing them from my PATH. Then reinstallning python. 

Thanks alot for this!

/Johan
